state.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"net/http"
+	"sync"
+	"time"
+)
+
+type state struct {
+	sync.RWMutex
+	client *http.Client
+}
+
+var proxyStateInstance state
+
+func GetProxyStateInstance() *state {
```

Comment:
Should we just put this state in a new "Proxy" type, and have the `proxyHandler` function be a function on that type? I'm not a big fan of global singletons.